imap_processing/cdfutils/global_base.py

```diff
@@ -0,0 +1,353 @@
+        The values for filling data
+    scale_type: str, default="linear"
+        The scale of the axis, "SCALETYP" attribute
+    label_axis: str, default=None
+        Axis label, "LABLAXIS" attribute
+    format: str, default=None
+        The format of the data, in Fortran format
+    units: str, default=None
+        The units of the data
+    """
+
+    validmin: np.float64 | np.int64
+    validmax: np.float64 | np.int64
+    display_type: str = None
+    catdesc: str = None
+    fieldname: str = None
+    var_type: str = "support_data"
+    fill_val: np.int64 = GlobalConstants.INT_FILLVAL
+    scale_type: str = "linear"
+    label_axis: str = None
+    format: str = None
+    units: str = None
+
+    def output(self):
+        """
+        Generate the output for the data level as a dictionary.
+
+        Returns
+        -------
+        Dictionary of correctly formatted values for the attributes in the class
+        """
+        return {
+            "CATDESC": self.catdesc,
+            "DISPLAY_TYPE": self.display_type,
+            "FIELDNAM": self.fieldname,
+            "FILLVAL": GlobalConstants.INT_FILLVAL,
+            "FORMAT": self.format,
+            "LABLAXIS": self.label_axis,
+            "UNITS": self.units,
+            "VALIDMIN": self.validmin,
+            "VALIDMAX": self.validmax,
+            "VAR_TYPE": self.var_type,
+            "SCALETYP": self.scale_type,
+        }
+
+
+@dataclass
+class ScienceBase(TypeBase):
+    """
+    The class for Science attributes, in particular with depend_0 attributes.
+
+    It also contains all the attributes and defaults from the generic TypeBase as well.
+
+    Attributes
+    ----------
+    depend_0: str = None
+        The first degree of dependent coordinate variables.
+        Although this is an optional keyword, it is required for every instance.
+    depend_1: str = None, optional
+        The second degree of dependent coordinate variables. This is used for 2d data.
+    depend_2: str = None, optional
+        The third degree of dependent coordinate variables. This is used for 3d data.
+        If this variable is used, there must also be a depend_1 value.
+    variable_purpose: str = None, optional
+        The variable purpose attribute tells which variables are worth plotting.
+    var_notes: str = None, optional
+        Notes on the variable
+    """
+
+    depend_0: str = None
+    depend_1: str = None
+    depend_2: str = None
+    variable_purpose: str = None
+    var_notes: str = None
+
+    def output(self):
+        """
+        Generate the output for the data level as a dictionary.
+
+        Returns
+        -------
+        Dictionary of correctly formatted values for the attributes in the class.
+        If the optional parameters are not defined, they are not included as attributes
+        in the output.
+        """
+        if self.depend_0 is None:
+            raise TypeError("ScienceBase requires depend_0 attribute.")
```

Comment:
Could these checks be done in `__post_init__`?
https://docs.python.org/3/library/dataclasses.html#dataclasses.__post_init__